text/metrics/0146-metrics-prototype-scenarios.md

```diff
@@ -0,0 +1,222 @@
+* How to configure the SDK and exporter
+
+Considering there might be multiple grocery stores, the metrics we collect will
+have the store name as a dimension - which is fairly static (not changing while
+the store is running).
+
+The store has plenty supply of potato and tomato, with the following price:
+
+* Potato: $1.00 / ea
+* Tomato: $3.00 / ea
+
+Each customer has a unique name (e.g. customerA, customerB), a customer could
+come to the store multiple times. Here goes the Python snippet:
+
+```python
+store = GroceryStore("Portland")
+store.process_order("customerA", {"potato": 2, "tomato": 3})
+store.process_order("customerB", {"tomato": 10})
+store.process_order("customerC", {"potato": 2})
+store.process_order("customerA", {"tomato": 1})
+```
+
+We will need the following metrics every minute:
+
+**Order info:**
+
+| Store    | Customer  | Number of Orders | Amount (USD) |
+| -------- | --------- | ---------------- | ------------ |
+| Portland | customerA | 2                | 14.00        |
+| Portland | customerB | 1                | 30.00        |
+| Portland | customerC | 1                | 2.00         |
+
+**Items sold:**
+
+| Store    | Customer  | Item   | Count |
+| -------- | --------- | ------ | ----- |
+| Portland | customerA | potato | 2     |
+| Portland | customerA | tomato | 4     |
+| Portland | customerB | tomato | 10    |
+| Portland | customerC | potato | 2     |
+
+## Scenario 2: HTTP Server
+
+The _HTTP Server_ scenario covers how a library developer X could use metrics
+API to instrument a library, and how the application developer Y can configure
+the library to use OpenTelemetry SDK in a final application. X and Y are working
+for different companies and they don't communicate. The demo has two parts - the
+library (HTTP lib owned by X) and the server app (owned by Y):
+
+* How developer X could instrument the library code in a vendor agnostic way
+  * Performance is critical for X
+  * X doesn't know which metrics and which dimensions will Y pick
+  * X doesn't know the aggregation time window, nor the final destination of the
+    metrics
+  * X would like to provide some default recommendation (e.g. default
+    dimensions, aggregation time window, histogram buckets) so consumers of his
+    library can have a better onboarding experience.
+* How developer Y could configure the SDK and exporter
+  * How should Y hook up the metrics SDK with the library
+  * How should Y configure the time window(s) and destination(s)
+  * How should Y pick the metrics and the dimensions
+
+### Library Requirements
+
+The library developer (developer X) will provide two libraries:
+
+* Server climate control library - a library which monitors and controls the
+  temperature and humidity of the server.
+* HTTP server library - a library which provides HTTP service.
+
+Both libraries will provide out-of-box metrics, the metrics have two categories:
+
+* Push metrics - the value is reported (via the API) when it is available, and
```

Comment:
nit: Do you want to clarify which metric in the example is Push/Pull? (I assume climate one only has Pull ones?)

Reply:
I think push/pull metrics can be implementation details. For example, the HTTP server duration can be implemented as either push or pull.

Reply:
If later we feel there is a need to make it concrete, we can have a small follow up PR.